### PR TITLE
[feat] Customized Icons for Renewable Energy Technology and Status (incl. Proposed COD)

### PR DESCRIPTION
**Pull request by nehaahussain**
### Description
[//]: # "[Required] Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- Added the icon files for status and technology tags to the assets folder.
- Created status tag component that displays the status with the status icon.
- Created technology tag component that displays the technology with the technology icon.

### Screenshots
[//]: # "Add any screenshots that will help your PR reviewer understand what’s going on in the PR!"
[//]: # "[Required] For frontend tasks: add screenshots of the layout and expected behavior (GIFs if you’re feeling extra fancy)!"
[//]: # "Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 
Screenshot 2024-11-08 at 3 35 55 PM

## How to review
[//]: # "/[Required] Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"

## Next steps
[//]: # "[Optional] - What has not yet been added to the PR, doesn’t yet have any functionality — essentially, what needs to be fixed or added."
[//]: # "[Optional] any temporary fixes in this PR that should be cleaned up later."
- battery storage vs energy storage?
- change structure of components so we can reuse later? how

### Online sources
[//]: # "[Optional] insert links to any useful tutorials or documentation that you used for this PR!"

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @itsliterallymonique

### CODE REVIEW

File: assets/Technology-Tag-Icons/icons.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
+export const OffshoreWindIcon = () => (
```
Comment (itsliterallymonique) [inline assets/Technology-Tag-Icons/icons.tsx:1]: for all technology tag icons, add the following props `props: { fill: string | undefined; stroke: string | undefined; }`. This will allow us to change the color of the SVG (so that Ethan can change them to navy!). Then for all fill and stroke (except for when fill="none"), use the props e.g. fill={props.fill}. I will comment where you should do that for the first 2 icons! Below is a screenshot to help.

Screenshot 2024-11-11 at 11 00 11 AM

File: assets/Technology-Tag-Icons/icons.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      fill="#4974E0"
```
Comment (itsliterallymonique) [inline assets/Technology-Tag-Icons/icons.tsx:13]: `fill={props.fill}`

File: assets/Technology-Tag-Icons/icons.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+      stroke="#2E3A59"
```
Comment (itsliterallymonique) [inline assets/Technology-Tag-Icons/icons.tsx:17]: `stroke={props.stroke}`

File: assets/Technology-Tag-Icons/icons.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+      fill="#4974E0"
```
Comment (itsliterallymonique) [inline assets/Technology-Tag-Icons/icons.tsx:22]: `fill={props.fill}`

File: assets/Technology-Tag-Icons/icons.tsx (hunk 2 above)
Comment (itsliterallymonique) [inline assets/Technology-Tag-Icons/icons.tsx:13]: Then for all the fill and stroke colors, make sure to add them to the colors.ts global colors stylesheet if they are not there already!

File: assets/Technology-Tag-Icons/icons.tsx [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    />
+  </svg>
+);
+
+export const EnergyStorageIcon = () => (
```
Comment (itsliterallymonique) [inline assets/Technology-Tag-Icons/icons.tsx:27]: `export const EnergyStorageIcon = (props: { fill: string | undefined; stroke: string | undefined; })` etc...

File: assets/Technology-Tag-Icons/icons.tsx [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+  <svg
+    width="13"
+    height="9"
+    viewBox="0 0 13 9"
+    fill="none"
+    xmlns="http://www.w3.org/2000/svg"
+  >
+    <path
+      d="M11.9228 6.4H11.8228V6.5C11.8228 7.00391 11.6226 7.48718 11.2663 7.8435C10.91 8.19982 10.4267 8.4 9.92278 8.4H2.2085C1.70458 8.4 1.22131 8.19982 0.864993 7.8435C0.508674 7.48718 0.308496 7.00391 0.308496 6.5V2.5C0.308496 1.99609 0.508674 1.51282 0.864993 1.1565C1.22131 0.800178 1.70458 0.6 2.2085 0.6H9.92278C10.4267 0.6 10.91 0.800178 11.2663 1.1565C11.6226 1.51282 11.8228 1.99609 11.8228 2.5V2.6H11.9228C12.0731 2.6 12.2172 2.6597 12.3235 2.76597C12.4297 2.87224 12.4894 3.01638 12.4894 3.16667V5.83333C12.4894 5.98362 12.4297 6.12776 12.3235 6.23403C12.2172 6.3403 12.0731 6.4 11.9228 6.4ZM9.92278 1.73333H2.2085C2.00516 1.73333 1.81016 1.81411 1.66638 1.95789C1.5226 2.10166 1.44183 2.29667 1.44183 2.5V6.5C1.44183 6.70333 1.5226 6.89834 1.66638 7.04212C1.81016 7.18589 2.00516 7.26667 2.2085 7.26667H9.92278C10.1261 7.26667 10.3211 7.18589 10.4649 7.04212C10.6087 6.89834 10.6894 6.70333 10.6894 6.5V2.5C10.6894 2.29667 10.6087 2.10166 10.4649 1.95788C10.3211 1.81411 10.1261 1.73333 9.92278 1.73333Z"
+      fill="#4896BC"
```
Comment (itsliterallymonique) [inline assets/Technology-Tag-Icons/icons.tsx:37]: `fill={props.fill}`

File: assets/Technology-Tag-Icons/icons.tsx [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+      stroke="white"
```
Comment (itsliterallymonique) [inline assets/Technology-Tag-Icons/icons.tsx:38]: stroke={props.stroke}

File: components/TechnologyTag/index.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+import { TagText1 } from '../../styles/texts';
+import { TechnologyTagStyles } from './styles';
+
+export default function TechnologyTag({
+  technology,
+}: {
+  technology: string | undefined;
+}) {
+  const iconMap: { [key: string]: JSX.Element } = {
+    'Solar PV': <SolarPvIcon />,
```
Comment (itsliterallymonique) [inline components/TechnologyTag/index.tsx:19]: For each SVG icon component, you will now need to pass in the fill and stroke props. Below is an example using Offshore Wind (but just make sure you use the colors.ts global colors stylesheet. it should look something like fill={COLORS.navy})

Screenshot 2024-11-11 at 11 03 35 AM

**Review by itsliterallymonique — CHANGES_REQUESTED** [review]
Please make the changes to the Techology-Tag-Icons SVG and your TechnologyTag that I have commented ASAP! Once you've made these changes, please let us know in the SVG thread on slack.

I will continue checking the rest of your PR soon and make any additional comments if I need to.

**You don't need to change anything with Status-Tag-Icons**

File: assets/Technology-Tag-Icons/icons.tsx [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+      fill={props.fill}
+    />
+  </svg>
+);
+
+export const LandBasedWindIcon = (props: {
+  fill: string | undefined;
+  stroke: string | undefined;
```
Comment (itsliterallymonique) [inline assets/Technology-Tag-Icons/icons.tsx:94]: if **stroke** is not passed into this icon, we don't need to have a stroke prop! You can remove this.

**Review by itsliterallymonique — APPROVED** [review]
You don't need to have a stroke prop for those icons that don't have a stroke in their path. But other than that, you can squash and merge when you're ready!
